core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncResponseTransformer.java

```diff
@@ -71,9 +72,26 @@ public void onResponse(ResponseT response) {
 
     @Override
     public void onStream(SdkPublisher<ByteBuffer> publisher) {
-        // onStream may be called multiple times so reset the file channel every time
-        this.fileChannel = invokeSafely(() -> createChannel(path));
-        publisher.subscribe(new FileSubscriber(this.fileChannel, path, cf, this::exceptionOccurred));
+        try {
+            createParentDirectoriesIfNeeded(path);
```

Comment:
We will probably have to clean up directories if it fails. Will update the PR

Reply:
It seems like it would be non-trivial to determine how many levels of parent directories were created and thus need to be cleaned up?